app/main.py

```diff
@@ -1,3 +1,69 @@
+from typing import Callable, Any
+
+
+def validate_operand(func: Callable) -> Callable:
+    def wrapper(self: "Distance", other: ("Distance", int, float)) -> Any:
+        if isinstance(other, Distance):
+            return func(self, other.km)
+        elif isinstance(other, (int, float)):
+            return func(self, other)
+        else:
+            raise TypeError(f"Unsupported operand type for {func.__name__}")
+
+    return wrapper
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: (int, float)) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    @validate_operand
+    def __add__(self, other: ("Distance", int, float)) -> "Distance":
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError("Unsupported operand type for +=")
+        return self
+
+    def __mul__(self, other: (int, float)) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        else:
+            raise TypeError("Unsupported operand type for *")
+
+    def __truediv__(self, other: int) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km / other, 2))
+        else:
+            raise TypeError("Unsupported operand type for /")
+
+    @validate_operand
+    def __lt__(self, other: ("Distance", int, float)) -> bool:
+        return self.km < other
+
+    @validate_operand
+    def __gt__(self, other: ("Distance", int, float)) -> bool:
+        return self.km > other
+
+    @validate_operand
+    def __eq__(self, other: ("Distance", int, float)) -> bool:
+        return self.km == other
+
+    @validate_operand
+    def __le__(self, other: ("Distance", int, float)) -> bool:
+        return self.km <= other
+
+    @validate_operand
+    def __ge__(self, other: ("Distance", int, float)) -> bool:
+        return self.km >= other
```

Comment:
no need to use validation here, u can use previouse created methods here
`hint: >= is not <`

Reply:
Done